### PR TITLE
Cumulus 1896

### PR DESCRIPTION
**Pull request by Jkovarik**
This PR updates all of the node.js lambdas to utilize cma-js 1.2.0, and makes them async (as the client library now requires this), and updates the CMA to 1.3.0 (which includes the streaming interface from this ticket).

It also includes a few performance tweaks from the profiling ticket last PI. 

See the CHANGELOG/migration for what/what/etc.

### CODE REVIEW

File: tf-modules/ingest/hello-world-task.tf [hunk 1]
```diff
@@ -6,7 +6,7 @@ resource "aws_lambda_function" "hello_world_task" {
   role             = var.lambda_processing_role_arn
   runtime          = "nodejs10.x"
   timeout          = 300
-  memory_size      = 256
+  memory_size      = 512
```
Comment (markdboyd) [inline tf-modules/ingest/hello-world-task.tf:9]: Why did the memory needs for our hello world task double?

Comment (Jkovarik) [reply tf-modules/ingest/hello-world-task.tf:9]: For the same reasons as https://github.com/nasa/cumulus/pull/1578 - upping memory updates processor power, the savings is .... roughly? 1:1. 

There's room for a discussion of tuning/risk profile rewards here, for sure.

File: bamboo/bootstrap-tf-deployment.sh [hunk 2]
```diff
@@ -58,10 +58,10 @@ echo "terraform {
 
 if [[ $NGAP_ENV = "SIT" ]]; then
   BASE_VAR_FILE="sit.tfvars"
-  CMA_LAYER_VERSION=7
+  CMA_LAYER_VERSION=10
 else
   BASE_VAR_FILE="sandbox.tfvars"
-  CMA_LAYER_VERSION=12
+  CMA_LAYER_VERSION=15
```
Comment (Jkovarik) [inline bamboo/bootstrap-tf-deployment.sh:64]: These update the CMA CI to v1.3.0

**Comment by Jkovarik** [comment]
Marking blocked as this is from the IP sprint

File: bamboo/bootstrap-tf-deployment.sh [hunk 3]
```diff
@@ -61,7 +61,7 @@ if [[ $NGAP_ENV = "SIT" ]]; then
   CMA_LAYER_VERSION=11
```
Comment (Jkovarik) [inline bamboo/bootstrap-tf-deployment.sh:61]: Todo:  This was removed from SIT yesterday, I need to re-add and update it in SIT.

Comment (Jkovarik) [reply bamboo/bootstrap-tf-deployment.sh]: 4249b8357

File: CHANGELOG.md [hunk 4]
```diff
@@ -13,6 +13,14 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
   3. [Re-deploy your `data-persistence` module](https://nasa.github.io/cumulus/docs/deployment/upgrade-readme#update-data-persistence-resources), which should re-create the `<prefix>-AccessTokensTable` DynamoDB table
   4. Return to using the Cumulus API/dashboard as normal
+- This release requires the Cumulus Message Adapter layer deployed with Cumulus Core to be at least 1.3.0, as the core lambdas have updated to [cumulus-message-adapter-js v1.2.0](https://github.com/nasa/cumulus-message-adapter-js/releases/tag/v1.2.0) and the new CMA interface.  As a result, users should:
+  1. Follow the [Cumulus Message Adapter(CMA) deployment instructions](https://nasa.github.io/cumulus/docs/deployment/deployment-readme#deploy-the-cumulus-message-adapter-layer) and install a CMA layer version >=1.3.0
+  2. If utilizing node.js in your workflows *and* the Cumulus CMA layer/`cumulus-message-adapter-js`, you must update your lambda to use [cumulus-message-adapter-js v1.2.0](https://github.com/nasa/cumulus-message-adapter-js/releases/tag/v1.2.0) and follow the migration instructions in the release notes.    Prior versions of `cumulus-message-adapter-js` are not compatible with CMA >= 1.3.0.
```
Comment (markdboyd) [inline CHANGELOG.md:18]:
```suggestion
  2. If you are using any custom Node.js Lambdas in your workflows **and** the Cumulus CMA layer/`cumulus-message-adapter-js`, you must update your lambda to use [cumulus-message-adapter-js v1.2.0](https://github.com/nasa/cumulus-message-adapter-js/releases/tag/v1.2.0) and follow the migration instructions in the release notes. Prior versions of `cumulus-message-adapter-js` are not compatible with CMA >= 1.3.0.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -13,6 +13,14 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
   2. Use the AWS console or CLI to delete your `<prefix>-AccessTokensTable` DynamoDB table
   3. [Re-deploy your `data-persistence` module](https://nasa.github.io/cumulus/docs/deployment/upgrade-readme#update-data-persistence-resources), which should re-create the `<prefix>-AccessTokensTable` DynamoDB table
   4. Return to using the Cumulus API/dashboard as normal
+- This release requires the Cumulus Message Adapter layer deployed with Cumulus Core to be at least 1.3.0, as the core lambdas have updated to [cumulus-message-adapter-js v1.2.0](https://github.com/nasa/cumulus-message-adapter-js/releases/tag/v1.2.0) and the new CMA interface.  As a result, users should:
+  1. Follow the [Cumulus Message Adapter(CMA) deployment instructions](https://nasa.github.io/cumulus/docs/deployment/deployment-readme#deploy-the-cumulus-message-adapter-layer) and install a CMA layer version >=1.3.0
```
Comment (markdboyd) [inline CHANGELOG.md:17]:
```suggestion
  1. Follow the [Cumulus Message Adapter (CMA) deployment instructions](https://nasa.github.io/cumulus/docs/deployment/deployment-readme#deploy-the-cumulus-message-adapter-layer) and install a CMA layer version >=1.3.0
```

File: example/cumulus-tf/python-reference-task.tf [hunk 6]
```diff
@@ -6,7 +6,7 @@ resource "aws_lambda_function" "python_reference_task" {
   role             = module.cumulus.lambda_processing_role_arn
   runtime          = "python3.7"
   timeout          = 300
-  memory_size      = 256
+  memory_size      = 1556
```
Comment (markdboyd) [inline example/cumulus-tf/python-reference-task.tf:9]: So again, the idea here is that the extra memory improves performance and reduces execution time?

File: tasks/discover-granules/index.js [hunk 7]
```diff
@@ -285,12 +285,9 @@ const discoverGranules = async ({ config }) => {
  *
  * @param {Object} event - a Cumulus Message
  * @param {Object} context - an AWS Lambda context
- * @param {Function} callback - an AWS Lambda handler
  * @returns {undefined} - does not return a value
```
Comment (markdboyd) [inline tasks/discover-granules/index.js:288]: returns a Promise now

Comment (markdboyd) [reply tasks/discover-granules/index.js:288]: This is going to be true for all of these handlers

File: tasks/files-to-granules/index.js [hunk 8]
```diff
@@ -86,8 +86,8 @@ function filesToGranules(event) {
 }
 exports.filesToGranules = filesToGranules;
 
-function handler(event, context, callback) {
-  cumulusMessageAdapter.runCumulusTask(
+async function handler(event, context, callback) {
```
Comment (markdboyd) [inline tasks/files-to-granules/index.js:89]: The `callback` param here is now obsolete and should be removed, I think

File: tf-modules/ingest/post-to-cmr-task.tf [hunk 9]
```diff
@@ -15,7 +15,7 @@ resource "aws_lambda_function" "post_to_cmr_task" {
       CMR_ENVIRONMENT             = var.cmr_environment
       stackName                   = var.prefix
       system_bucket               = var.system_bucket
-      CUMULUS_MESSAGE_ADAPTER_DIR = "/opt"
+      CUMULUS_MESSAGE_ADAPTER_DIR = "/opt/"
```
Comment (markdboyd) [inline tf-modules/ingest/post-to-cmr-task.tf:18]: Is the trailing slash necessary? If so, that feels potentially brittle

File: example/cumulus-tf/python-reference-task.tf (hunk 6 above)
Comment (Jkovarik) [reply example/cumulus-tf/python-reference-task.tf:9]: More or less.  It gives it more horsepower, and the improvement is linear. 

Upside - tests run seconds faster. 
Downside - screwing up might cost a lot more.

File: tasks/discover-granules/index.js (hunk 7 above)
Comment (Jkovarik) [reply tasks/discover-granules/index.js:288]: Argh.  Yes, you're correct, I thought I'd fixed that.     Good catch.

Comment (Jkovarik) [reply tasks/discover-granules/index.js:288]: 478d33f08

Most of them could reference the schema/output.json.  That seemed sufficient.

Comment (Jkovarik) [reply tasks/discover-granules/index.js]: (and 44b449f08)

File: tf-modules/ingest/post-to-cmr-task.tf (hunk 9 above)
Comment (Jkovarik) [reply tf-modules/ingest/post-to-cmr-task.tf:18]: Not even a little bit.  This was just making it consistent with all the other task configurations.

**Pull request merged**